packages/legacy/core/App/screens/ContactDetails.tsx

```diff
@@ -79,25 +87,55 @@ const ContactDetails: React.FC<ContactDetailsProps> = ({ route }) => {
     setIsCredentialsRemoveModalDisplayed(false)
   }
 
+  const handleGoToRename = () => {
+    navigation.navigate(Screens.RenameContact, { connectionId })
+  }
+
+  const callGoToRename = useCallback(() => handleGoToRename(), [])
   const callOnRemove = useCallback(() => handleOnRemove(), [])
   const callSubmitRemove = useCallback(() => handleSubmitRemove(), [])
   const callCancelRemove = useCallback(() => handleCancelRemove(), [])
   const callGoToCredentials = useCallback(() => handleGoToCredentials(), [])
   const callCancelUnableToRemove = useCallback(() => handleCancelUnableRemove(), [])
 
+  const contactLabel = useMemo(
+    () =>
+      (connection?.id && store.preferences.alternateContactNames[connection.id]) ||
+      connection?.alias ||
+      connection?.theirLabel,
+    [connection, store.preferences.alternateContactNames]
+  )
+
   return (
     <SafeAreaView style={{ flexGrow: 1 }} edges={['bottom', 'left', 'right']}>
-      {record({
-        fields: [
-          {
-            name: connection?.alias || connection?.theirLabel,
-            value: t('ContactDetails.DateOfConnection', {
-              date: connection?.createdAt ? formatTime(connection.createdAt, { includeHour: true }) : '',
-            }),
-          },
-        ] as Attribute[],
-        footer: () => <RecordRemove onRemove={callOnRemove} />,
-      })}
+      <View style={styles.contentContainer}>
+        <Text style={{ ...TextTheme.headingThree }}>{contactLabel}</Text>
+        <Text style={{ ...TextTheme.normal, marginTop: 20 }}>
+          {t('ContactDetails.DateOfConnection', {
+            date: connection?.createdAt ? formatTime(connection.createdAt, { includeHour: true }) : '',
+          })}
+        </Text>
+      </View>
+      <TouchableOpacity
```

Comment:
We should decide, perhaps at a Bifold meeting, if we roll with `TouchableOpacity` or `Pressable`.

Reply:
Ok. For now I'm gonna go with `Touchable` because it has the visible reactivity that is needed by default